llvm/test/Analysis/ScalarEvolution/trip-count-unknown-stride.ll

```diff
@@ -329,10 +329,9 @@ define void @ne_nsw_nonneg_step(ptr nocapture %A, i32 %n, i32 %s) mustprogress {
 ;
 ; CHECK-LABEL: 'ne_nsw_nonneg_step'
 ; CHECK-NEXT:  Determining loop execution counts for: @ne_nsw_nonneg_step
-; CHECK-NEXT:  Loop %for.body: backedge-taken count is (((-1 * %s) + %n) /u %s)
-; CHECK-NEXT:  Loop %for.body: constant max backedge-taken count is i32 -1
-; CHECK-NEXT:  Loop %for.body: symbolic max backedge-taken count is (((-1 * %s) + %n) /u %s)
-; CHECK-NEXT:  Loop %for.body: Trip multiple is 1
+; CHECK-NEXT:  Loop %for.body: Unpredictable backedge-taken count.
+; CHECK-NEXT:  Loop %for.body: Unpredictable constant max backedge-taken count.
+; CHECK-NEXT:  Loop %for.body: Unpredictable symbolic max backedge-taken count.
```

Comment:
Could probably recover these if we used loop guards for start.

Reply:
In this case we cannot recover these since the start value is zero :(
I will check if using `applyLoopGuards(Start, Guards)` is profitable.

Reply:
Wouldn't the start value after subtraction here be something like `-n`?

Reply:
Oh, sorry, the start value is `(zext i1 %tobool2.not to i32)`.

Reply:
https://github.com/dtcxzyw/llvm-opt-benchmark/pull/2212/files Looks like it is unprofitable.